run_configs/lfric/lfric_common.py

```diff
@@ -21,79 +23,113 @@ def check_available(self):
     logger.info('GenerateLoader')
+    names = [name.strip() for name in
+             open(config_dir / 'config_namelists.txt').readlines()]
+    configuration_mod_fpath = config_dir / 'configuration_mod.f90'
     gen_loader = Script(gen_loader_tool)
-    names = [name.strip() for name in open(config_dir / 'config_namelists.txt').readlines()]
-    configuration_mod_fpath = config.source_root / 'configuration_mod.f90'
     gen_loader.run(additional_parameters=[configuration_mod_fpath,
                                           *names])
 
     # create feign_config_mod.f90 in source root
+    # ------------------------------------------
     logger.info('GenerateFeigns')
-    feign_config = Script(gen_feigns_tool)
-    feign_config_mod_fpath = config.source_root / 'feign_config_mod.f90'
-    feign_config.run(additional_parameters=[str(config_dir / 'rose-meta.json'),
-                                            '-output', feign_config_mod_fpath])
+    feign_config_mod_fpath = config_dir / 'feign_config_mod.f90'
+    gft = Script(gen_feigns_tool)
+    gft.run(additional_parameters=[rose_meta,
+                                   '-output', feign_config_mod_fpath])
 
-    # put the generated source into an artefact
-    # todo: we shouldn't need to do this, should we?
-    #       it's just going to be found in the source folder with everything else.
-    config._artefact_store['configurator_output'] = [
-        configuration_mod_fpath,
-        feign_config_mod_fpath
-    ]
+    find_source_files(config, source_root=config_dir)
 
 
+# ============================================================================
 @step
 def fparser_workaround_stop_concatenation(config):
     """
-    fparser can't handle string concat in a stop statement. This step is a workaround.
+    fparser can't handle string concat in a stop statement. This step is
+    a workaround.
 
     https://github.com/stfc/fparser/issues/330
 
     """
-    feign_config_mod_fpath = config.source_root / 'feign_config_mod.f90'
+    feign_path = None
+    for file_path in config.artefact_store[ArtefactSet.FORTRAN_BUILD_FILES]:
+        if file_path.name == 'feign_config_mod.f90':
+            feign_path = file_path
+            break
+    else:
+        raise RuntimeError("Could not find 'feign_config_mod.f90'.")
 
     # rename "broken" version
-    broken_version = feign_config_mod_fpath.with_suffix('.broken')
-    shutil.move(feign_config_mod_fpath, broken_version)
+    broken_version = feign_path.with_suffix('.broken')
+    shutil.move(feign_path, broken_version)
 
     # make fixed version
     bad = "_config: '// &\n        'Unable to close temporary file'"
     good = "_config: Unable to close temporary file'"
 
-    open(feign_config_mod_fpath, 'wt').write(
+    open(feign_path, 'wt').write(
         open(broken_version, 'rt').read().replace(bad, good))
+
+
+# ============================================================================
+def get_transformation_script(fpath, config):
```

Comment:
Type hinting? Would remove the need for `rtype` in the docstring.

Reply:
Indeed.